Documentation update for 1.19 (#597)
HabanaAI/vllm-fork: README_GAUDI.md

- OS: Ubuntu 22.04 LTS
- Python: 3.10
- Intel Gaudi accelerator
- Intel Gaudi software version 1.19.0

## Quick start using Dockerfile

> ```

Reviewer: It needs more explanation.

Reviewer 2: Will you add it? @piotrbocian

Reviewer 2: Or explain what you have in mind so someone else can do it?

Reviewer 2: Other vendors seem to have it in the similar way as here:
https://docs.vllm.ai/en/latest/getting_started/openvino-installation.html#quick-start-using-dockerfile
https://docs.vllm.ai/en/latest/getting_started/cpu-installation.html#quick-start-using-dockerfile
https://docs.vllm.ai/en/latest/getting_started/arm-installation.html#quick-start-with-dockerfile
https://docs.vllm.ai/en/latest/getting_started/xpu-installation.html#quick-start-using-dockerfile

Reviewer: Please see how document is structured:

1. Quick start using Dockerfile
2. Build from source
 2.1 Environment verification
 2.2 Run Docker Image
 2.3 Build and Install vLLM

Questions:
- is (1.) full alternative to (2.)? If so, I would add one liner as
- "You can quickly set up vLLM using latest Intel Gaudi docker and vllm verson <TBD>"
- is (2.1 Env verification) common to (1.) and (2.)?

Use the following commands to run a Docker image:

```{.console}
$ docker pull vault.habana.ai/gaudi-docker/1.19.0/ubuntu22.04/habanalabs/pytorch-installer-2.5.1:latest
$ docker run -it --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none --cap-add=sys_nice --net=host --ipc=host vault.habana.ai/gaudi-docker/1.19.0/ubuntu22.04/habanalabs/pytorch-installer-2.5.1:latest
```

### Build and Install vLLM

> Currently, we are providing multiple repositories which can be used to install vLLM with Intel® Gaudi®, pick one option:

Reviewer: "multiple ways"

> Currently, we are providing multiple repositories which can be used to install vLLM with Intel® Gaudi®, pick one option:

Reviewer: ```suggestion
Currently, we are providing multiple ways which can be used to install vLLM with Intel® Gaudi®, pick one option:
```

#### 1. Build and Install the stable version

> Periodically, we are releasing vLLM to allign with Intel® Gaudi® software releases. The stable version is released with a tagg, and supports fully validated features and performance optimizations in Gaudi's [vLLM-fork](https://github.com/HabanaAI/vllm-fork). To install the stable release from [HabanaAI/vLLM-fork](https://github.com/HabanaAI/vllm-fork), run the following:

Reviewer 3: typo: tagg

Author: Corrected

```{.console}
$ git clone https://github.com/HabanaAI/vllm-fork.git
$ cd vllm-fork

> $ git checkout v1.19.0

Reviewer 4: Probably will need to replace with proper tag like in v0.5.3.post1+Gaudi-1.18.0

Author: @bartekkuncer please verify if that makes sense.

Reviewer 2: @michalkuligowski makes a good point especially that in release note we provide instruction with the use of a tag, so this change will make these two consistent.

$ pip install -r requirements-hpu.txt
$ python setup.py develop
```

> #### 3. Build and Install from vLLM main source

Reviewer: "from vLLM upstream"

> #### 3. Build and Install from vLLM main source

Reviewer: ```suggestion
#### 3. Build and Install from vLLM upstream
```

| 1                  | 1               | PyTorch lazy mode  |

> [!WARNING] 

> > In 1.19.0, all modes utilizing `PT_HPU_LAZY_MODE=0` are highly experimental and should be only used for validating functional correctness. Their performance will be improved in the next releases. For obtaining the best performance in 1.19.0, please use HPU Graphs, or PyTorch lazy mode.

Reviewer: ```suggestion
> In current release, all modes utilizing `PT_HPU_LAZY_MODE=0` are highly experimental and should be only used for validating functional correctness. Their performance will be improved in the next releases. For obtaining the best performance please use HPU Graphs, or PyTorch lazy mode.
```
